osu.Game/Overlays/Settings/Sections/Graphics/LayoutSettings.cs

```diff
@@ -140,15 +149,33 @@ private void load(FrameworkConfigManager config, OsuConfigManager osuConfig, Gam
                     Bindable = sizeFullscreen
                 };
 
+                sizeSettingsContainer.Child = windowSizeDropdown = new ResolutionSettingsDropdown
+                {
+                    LabelText = "Window size",
+                    ShowsDefaultIndicator = false,
+                    Items = resolutions,
+                    Bindable = windowedSize
+                };
+
                 windowModeDropdown.Bindable.BindValueChanged(mode =>
                 {
                     if (mode.NewValue == WindowMode.Fullscreen)
                     {
                         resolutionDropdown.Show();
+                        windowSizeDropdown.Hide();
                         sizeFullscreen.TriggerChange();
                     }
+                    else if (mode.NewValue == WindowMode.Windowed)
+                    {
+                        windowSizeDropdown.Show();
+                        resolutionDropdown.Hide();
+                        windowedSize.TriggerChange();
+                    }
                     else
+                    {
                         resolutionDropdown.Hide();
+                        windowSizeDropdown.Hide();
+                    }
```

Comment:
I'm not a fan of these three branches having their independent `Show()/Hide()` calls; that's bug-prone. Instead I'd probably do something like:

```csharp
resolutionDropdown.FadeTo(mode.NewValue == WindowMode.Fullscreen ? 1 : 0);
windowSizeDropdown.FadeTo(mode.NewValue == WindowMode.Windowed ? 1 : 0);
```

Unfortunately there's no good way to do the same thing with the `TriggerChange()` calls but I feel that would bring much readability and significantly de-branch this code.

Reply:
I'll get to that tomorrow. I'm guessing that you mean to fully remove the if else statement and change it to a conditional statement. Might it be possible to get a value from `resolutionDropdown.Visible` or something like get and put it in a `switch()` statement to trigger the `TriggerChange()` call?

Reply:
I'd keep the `TriggerChange()` calls in the if-else as they are right now, just hoist the dropdown visibility stuff outside of it and use the branchless versions above. Using dropdown visibility to trigger these bindable changes reads worse in my opinion.